bin/s2nd.c

```diff
@@ -166,8 +166,10 @@ int cache_store_callback(struct s2n_connection *conn, void *ctx, uint64_t ttl, c
 {
     struct session_cache_entry *cache = ctx;
 
-    S2N_ERROR_IF(key_size == 0 || key_size > MAX_KEY_LEN, S2N_ERR_INVALID_ARGUMENT);
-    S2N_ERROR_IF(value_size == 0 || value_size > MAX_VAL_LEN, S2N_ERR_INVALID_ARGUMENT);
+    POSIX_ENSURE(key_size > 0, S2N_ERR_INVALID_ARGUMENT);
+    POSIX_ENSURE(key_size <= MAX_KEY_LEN, S2N_ERR_INVALID_ARGUMENT);
+    POSIX_ENSURE(value_size > 0, S2N_ERR_INVALID_ARGUMENT);
+    POSIX_ENSURE(value_size <= MAX_VAL_LEN, S2N_ERR_INVALID_ARGUMENT);
```

Comment:
Why not?
```suggestion
    POSIX_ENSURE_INCLUSIVE_RANGE(1, key_size, MAX_KEY_LEN);
    POSIX_ENSURE_INCLUSIVE_RANGE(1, value_size, MAX_VAL_LEN);
```

If this doesn't work, does this indicate missing macros which should exist? (I'm guessing that the error code is incorrect. So, should POSIX_ENSURE_INCLUSIVE_RANGE accept an error code?)

Reply:
Yeah the limitation is it doesn't take an error code. But in this case it doesn't really matter so I can change it to that. We can address the comparison macros taking a code in another context.